### PR TITLE
Feature/refactoring endpoints

### PR DESCRIPTION
**Pull request by ThijsBouwes**
| Q | A |
|--|--|
| Status | READY |
| Migrations | YES  |

## Description
Refactored all the endpoints, so the namespacing and naming conventions are all the same.
Also added / updated the swagger docs for some endpoints.

## Steps to test/reproduce
1. Pull the changes
2. Run `php ./vendor/bin/codecept run`
3. Make swagger docs and test them `php artisan l5-swagger:generate`
4. Checkout the swagger docs at cavalry-api.dev/api/documentation

### CODE REVIEW

File: app/Http/Controllers/Api/Agent/CreateController.php [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-use Illuminate\Http\Request;
-use Illuminate\Http\Response;
+use App\Http\Controllers\Api\Controller;
+use App\Http\Requests\Agent\AgentCreateRequest;
 
 /**
  * @SWG\Post(
  *     path="/agents",
- *     tags={"agents"},
+ *     tags={"Agents"},
```
Comment (titopixelfusion) [inline app/Http/Controllers/Api/Agent/CreateController.php]: IMO tags should be lower case

File: app/Http/Controllers/Api/Backup/Config/CreateController.php [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 use App\Models\BackupConfig;
 use Illuminate\Http\JsonResponse;
+use App\Http\Controllers\Api\Controller;
+use App\Http\Requests\Backup\Config\CreateRequest;
 
 /**
  * @SWG\Post(
  *     path="/backups/configs",
- *     tags={"backups"},
+ *     tags={"BackupConfig"},
```
Comment (titopixelfusion) [inline app/Http/Controllers/Api/Backup/Config/CreateController.php]: should be multiple tags maybe, backup and config

File: routes/api.php [hunk 3]
```diff
@@ -18,31 +18,30 @@
...
 Route::group(['namespace' => 'Api\Agent', 'prefix' => 'agents'], function () {
     Route::get('/', 'ListController');
-    Route::get('/{agent}', 'FetchController')->where(['agent' => '[1-9][0-9]*']);
     Route::post('/', 'CreateController');
+    Route::get('/{agent}', 'FetchController')->where(['agent' => '[1-9][0-9]*']);
     Route::delete('/{agent}', 'DeleteController')->where(['agent' => '[1-9][0-9]*']);
 });
 
 // Backup jobs
-Route::group(['namespace' => 'Api\Backup\Job', 'prefix' => 'jobs'], function () {
```
Comment (titopixelfusion) [inline routes/api.php]: Should we have a group inside a group? So the main group would be for backups and then subgroups for job, messages and so on, this way we don't repeat the url prefix backups..

File: app/Http/Controllers/Api/Agent/CreateController.php (hunk 1 above)
Comment (ThijsBouwes) [inline app/Http/Controllers/Api/Agent/CreateController.php]: Done

File: app/Http/Controllers/Api/Backup/Config/CreateController.php (hunk 2 above)
Comment (ThijsBouwes) [inline app/Http/Controllers/Api/Backup/Config/CreateController.php]: Done, the endpoints show multiple times.
For example `{"backup", "config"}`, now the endpoint is listed under "backup" and "config".

File: routes/api.php (hunk 3 above)
Comment (ThijsBouwes) [inline routes/api.php]: I grouped them together, i think we should make a route group.
So we can prefix the version of our API.

File: app/Http/Controllers/Api/Backup/Destination/ListController.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?php
+namespace App\Http\Controllers\Api\Backup\Destination;
+
+use App\Models\BackupJob;
+use Illuminate\Http\JsonResponse;
```
Comment (jpcaparas) [inline app/Http/Controllers/Api/Backup/Destination/ListController.php]: Alphabetical arrangement of packages.

File: app/Http/Controllers/Api/Backup/Message/ListController.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ */
+class ListController extends Controller
+{
+    public function __invoke(BackupJob $backupJob)
+    {
+        $messages = $backupJob->backupJobMessages()->getResults();
+
+        return new JsonResponse($messages);
+    }
+}
```
Comment (jpcaparas) [inline app/Http/Controllers/Api/Backup/Message/ListController.php]: @ThijsBouwes would be a good practice to always leave a new line `\n` on every file created.

**Pull request merged**